### PR TITLE
Fixes relating to --build-on-remote

### PR DESCRIPTION
**Pull request by Mic92**
fixes https://github.com/nix-community/nixos-anywhere/pull/366

rebased version of the same

### CODE REVIEW

**Review by Enzime — APPROVED** [review]
LGTM

Just needs `nixfmt` to be run on each commit :+1:

File: terraform/install/run-nixos-anywhere.sh [hunk 1]
```diff
@@ -2,28 +2,36 @@
 set -euo pipefail
 
 SCRIPT_DIR="$(realpath "$(dirname "${BASH_SOURCE[0]}")")"
+
+declare -A input
+
+while IFS= read -r -d '' key && IFS= read -r -d '' value; do
+  input[$key]=$value
+done < <(jq -j 'to_entries[] | (.key, "\u0000", .value, "\u0000")' <<<$ARGUMENTS)
```
Comment (Enzime) [inline terraform/install/run-nixos-anywhere.sh:10]:
```suggestion
done < <(jq -j 'to_entries[] | (.key, "\u0000", .value, "\u0000")' <<<"$ARGUMENTS")
```

**Review by Enzime — CHANGES_REQUESTED** [review]
`shellcheck` failed :laughing:

File: terraform/install/run-nixos-anywhere.sh (hunk 1 above)
Comment (Mic92) [reply terraform/install/run-nixos-anywhere.sh:10]: shfmt does that...

Comment (Mic92) [reply terraform/install/run-nixos-anywhere.sh]: I found a workaround.

**Comment by Mic92** [comment]
@mergify queue

**Pull request merged**